ntoskrnl/lpc/complete.c

```diff
@@ -275,7 +278,8 @@ NtAcceptConnectPort(OUT PHANDLE PortHandle,
     if (ClientSectionToMap)
     {
         /* Setup the offset */
-        SectionOffset.QuadPart = ConnectMessage->ClientView.SectionOffset;
+        SectionOffset.LowPart = ConnectMessage->ClientView.SectionOffset;
```

Comment:
Why this change?

Reply:
The question I have, when I see this (and the other instances) is whether these changes break x64 compatibility (indeed in x64 an offset could theoretically be larger than what can be held in a DWORD.)

Reply:
This is just small compiler compatibility improvement. If the compiler does not have built-in support for 64-bit integers, you should always use the LowPart and HighPart quantities instead of QuadPart.

Reply:
Do we use any compilers that don't support QuadPart? If not, this is not needed, unless you have some plans on it (then please explain them).

Reply:
We don't support compilers without `long long` support, so please use such pattern only when needed by logic

Reply:
A ton of code in our code base requires this long long support be present already (and in other parts of the kernel), so there is no need to keep that.

Reply:
Microsoft defines LARGE_INTEGER in Windows NT as data type that is 64-bit integer created from an array of two longwords. It is not the same as quadword and this is how I used it.

Reply:
This is still true for at least NT 5.2. Maybe newer versions changes that, dunno. Anyway, will apply your guidelines in free time.

Reply:
Fixed.